### PR TITLE
NUTCH-2289 SSL support for Nutch 2.X REST API.

### PR DESCRIPTION
**Pull request by kamaci**

### CODE REVIEW

File: conf/nutch-default.xml [hunk 1]
```diff
@@ -1435,4 +1435,60 @@
   </description>
 </property>
 
+<property>
+  <name>restapi.auth</name>
+  <value>NONE</value>
+  <description>
+    Configures authentication type for communicating with RESTAPI. Valid values are BASIC, DIGEST, SSL and NONE.
+    When no authentication type is defined NONE will be used as default which does not provide security.
+    Use the restapi.auth.username and restapi.auth.auth.password properties to configure
```
Comment (lewismc) [inline conf/nutch-default.xml]: Typo again

File: src/java/org/apache/nutch/api/NutchServer.java [hunk 2]
```diff
@@ -99,17 +114,70 @@ public NutchServer() {
...
+      return;
+    }
+
+    String username = configManager.get(ConfigResource.DEFAULT).get("restapi.auth.username", "admin");
+    String password = configManager.get(ConfigResource.DEFAULT).get("restapi.auth.password", "nutch");
+
+    MapVerifier mapVerifier = new MapVerifier();
+    mapVerifier.getLocalSecrets().put(username, password.toCharArray());
+
+    if (authenticationType == AuthenticationTypeEnum.BASIC) {
```
Comment (lewismc) [inline src/java/org/apache/nutch/api/NutchServer.java]: Please see comment on other issue for use of switch block.

**Comment by lewismc** [comment]
PING @kamaci

**Comment by kamaci** [comment]
@lewismc I've fixed necessary changes at forked branch. I'll pull to this branch after it forked branch is approved.

**Comment by kamaci** [comment]
I'll create a new pull request which does not have conflicts.

**Pull request closed without merging**